forge-ai/src/main/java/forge/ai/SpellAbilityAi.java

```diff
@@ -220,30 +246,30 @@ protected boolean doTriggerAINoCost(final Player aiPlayer, final SpellAbility sa
                 if (sa.canTarget(p)) {
                     sa.resetTargets();
                     sa.getTargets().add(p);
-                    return true;
+                    return new AiAbilityDecision(100, AiPlayDecision.WillPlay);
                 }
             }
 
-            return false;
+            return new AiAbilityDecision(0, AiPlayDecision.TargetingFailed);
         }
-        return true;
+        return new AiAbilityDecision(0, AiPlayDecision.CantPlayAi);
```

Comment:
logic inverted
the problem is canPlayWithoutRestrict leads to checkApiLogic which when not defined for api only causes random success
this breaks something like a Static Cleanup trigger in an almost impossible to debug way